src/Engine/UnitMods/CountTracker.java

```diff
@@ -7,33 +7,35 @@
 {
   private static final long serialVersionUID = 1L;
 
-  private HashMap<RootKeyType, HashMap<InstanceKeyType, Integer>> buildCounts = new HashMap<>();
+  private HashMap<RootKeyType, HashMap<InstanceKeyType, Integer>> countMap = new HashMap<>();
 
-  public boolean hasCountFor(RootKeyType co)
+  public boolean hasCountFor(RootKeyType root)
   {
-    return buildCounts.containsKey(co);
+    return countMap.containsKey(root);
   }
-  public HashMap<InstanceKeyType, Integer> getCountFor(RootKeyType co)
+  public HashMap<InstanceKeyType, Integer> getCountFor(RootKeyType root)
   {
-    if( !buildCounts.containsKey(co) )
-      buildCounts.put(co, new HashMap<InstanceKeyType, Integer>());
-    return buildCounts.get(co);
+    if( !countMap.containsKey(root) )
+      countMap.put(root, new HashMap<InstanceKeyType, Integer>());
```

Comment:
Seems a bit odd that calling an accessor mutates the collection here, especially since the sibling `getCountFor` does not.

Reply:
This is specifically getting a count *collection* - if I returned `null` here, the caller would have to check for null and possibly instantiate one needlessly.
It's intended to be read/write access, too, so that's also relevant.

Reply:
> It's intended to be read/write access

As in, someone could add to the returned collection and have it reflected in the next call to `getCountFor`? Is it used this way anywhere? Is that tested?

I'll grant that it's probably better to create the empty collection due to user expectations, and since it shouldn't affect any future calls.

Reply:
Yes, no, no.
I don't really feel the need to write tests for `HashMap`. If that's broken, we have bigger fish to fry.

Reply:
Fair enough. I guess you are returning a pointer to a hashmap, so any changes will be reflected back at the mothership.
I'm not sure I'd encourage that kind of use, but that's neither here nor there.